### PR TITLE
add import for tims-autoMSMS acquisitions on timsTOF classic

### PR DESCRIPTION
**Pull request by SteffenHeu**
internally handled like a pasef file with a single precursor, which gives you the option to resolve coeluting isomers by mobility resolving

### CODE REVIEW

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/TDFImportTask.java [hunk 1]
```diff
@@ -547,6 +548,37 @@ private void assignBbCidMsMsInfo(IMSRawDataFile newMZmineFile, TDFFrameTable fra
...
+
+    final int firstFrameId = (int) frameTable.getFirstFrameNumber();
+
+    for (int i = 0; i < frames.size(); i++) {
+      Frame frame = frames.get(i);
+      final int frameTableIndex = frame.getFrameId() - firstFrameId;
+
+      if (frameTable.getScanModeColumn().get(frameTableIndex).intValue()
+          != BrukerScanMode.AUTO_MSMS.getNum()
+          || frameTable.getMsMsTypeColumn().get(frameTableIndex).intValue() != 2) {
```
Comment (robinschmid) [inline mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/TDFImportTask.java:566]: is this the MS level or what does this refer to?
`frameTable.getMsMsTypeColumn().get(frameTableIndex).intValue() != 2`

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/TDFImportTask.java [hunk 2]
```diff
@@ -547,6 +548,37 @@ private void assignBbCidMsMsInfo(IMSRawDataFile newMZmineFile, TDFFrameTable fra
...
+        continue;
+      }
+
+      final int frameMsMsTableIndex = BinarySearch.binarySearch(frameMsMsInfoTable.getFrameId(),
+          (double) frame.getFrameId(), DefaultTo.MINUS_INSERTION_POINT, Long::doubleValue);
```
Comment (robinschmid) [inline mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/TDFImportTask.java:571]: not needed here but maybe we should also add a long version to binarySearch to not have to cast to double

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/datamodel/sql/TDFFrameMsMsInfoTable.java [hunk 3]
```diff
@@ -112,6 +116,16 @@ public DDAMsMsInfo getDDAMsMsInfo(int index, int msLevel, Scan msmsScan, Scan pa
     return ddaMsMsInfo;
   }
 
+  public PasefMsMsInfo getImsAutoMsMsInfo(int index, Frame msmsScan, Frame parentScan) {
+    final double precursor = precursorMz.get(index).doubleValue();
+    final double width = isolationWidth.get(index).doubleValue();
+    final float ce = getCe().get(index).floatValue();
```
Comment (robinschmid) [inline mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/datamodel/sql/TDFFrameMsMsInfoTable.java:122]: those values are always not null? or add null checks?
PasefMsMsInfoImpl also uses Float so could be used here as well.

**Review by robinschmid — APPROVED** [review]

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/datamodel/sql/TDFFrameMsMsInfoTable.java (hunk 3 above)
Comment (SteffenHeu) [reply mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/datamodel/sql/TDFFrameMsMsInfoTable.java:122]: never seen null values here, they have always been set. the same is assumed in the other methods that build msms infos from bruker files

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/TDFImportTask.java (hunk 1 above)
Comment (SteffenHeu) [reply mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/TDFImportTask.java:566]: see javadoc in the TDFFrame table
grafik

Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/TDFImportTask.java:566]: ah ok.

**Pull request merged**